app/models/media/object.rb

```diff
@@ -7,12 +7,25 @@ class Media::Object < ActiveRecord::Base
   delegate :mission, to: :answer
   delegate :url, to: :item, allow_nil: true
 
+  before_save :generate_token
+
   scope :expired, -> { where(answer_id: nil).where("created_at < ?", 12.hours.ago) }
 
   def download_url
     dl_url = url
     separator = (dl_url =~ /\?/) ? "&" : "?"
-    "#{dl_url}#{separator}dl=1"
+    "#{dl_url}#{separator}token=#{token}&dl=1"
+  end
+
+  def token_url(style: nil)
```

Comment:
Can there be a comment on this method explaining why this is necessary. The bug that necessitated this feature obviously wasn't obvious to us initially, so it would be good to capture the reasoning here.